### PR TITLE
Add support of initial arguments

### PR DESCRIPTION
**Pull request by oltarasenko**
Implemented as a part of #126 feature request

### CODE REVIEW

File: lib/crawly/engine.ex [hunk 1]
```diff
@@ -22,12 +22,52 @@ defmodule Crawly.Engine do
...
+             result:
+               :ok
+               | {:error, :spider_already_started}
+               | {:error, :atom}
+
+  def start_spider(spider_name, options) when is_list(options) do
+    start_spider(spider_name, UUID.uuid1(), options)
+  end
+
+  @spec start_spider(spider_name, crawl_id, options) :: result
```
[the remark this answers is not in the document]
Comment (Ziinc) [reply lib/crawly/engine.ex:60]: Something like:
```elixir
def start_spider(spider_name, opts \\ []) do
  opts = 
        Enum.into(%{})
        |> Map.put_lazy(:crawl_id, &UUID.uuid/1)
  
end
```

[the remark this answers is not in the document]
Comment (Ziinc) [reply lib/crawly/engine.ex:60]: would probably be good to add typespecs for the options too

File: lib/crawly/manager.ex [hunk 2]
```diff
@@ -45,15 +45,15 @@ defmodule Crawly.Manager do
     end
   end
 
-  def start_link(spider_name) do
+  def start_link([spider_name, options]) do
```
Comment (Ziinc) [inline lib/crawly/manager.ex:48]: I think a tuple would be more appropriate compared to a list

**Review by Ziinc — CHANGES_REQUESTED** [review]
engine implementation and tests looks great, the `start_spider/2` interface can be cleaner

**Review by Ziinc — APPROVED** [review]

**Comment by Ziinc** [comment]
Have a glance at my changes, merge in if ok.

**Review by oltarasenko — COMMENTED** [review]
Thanks for improving it. That looks fine from my point of view. Merging.

**Pull request merged**